exo/main.py

```diff
@@ -56,6 +59,15 @@
 
 shard_downloader: ShardDownloader = HFShardDownloader(quick_check=args.download_quick_check, max_parallel_downloads=args.max_parallel_downloads)
 inference_engine_name = args.inference_engine or ("mlx" if system_info == "Apple Silicon Mac" else "tinygrad")
+print(f"Inference engine name after selection: {inference_engine_name}")
+
+if inference_engine_name not in ["mlx", "tinygrad", "dummy"]:
+  print(f"Warning: Unknown inference engine '{inference_engine_name}'. Defaulting to 'tinygrad'.")
+  inference_engine_name = "tinygrad"
+else:
+  print(f"Using selected inference engine: {inference_engine_name}")
+
+print(f"About to call get_inference_engine with: {inference_engine_name}")
```

Comment:
No need for any of this. If they choose to override the inference engine name with an unsupported inference engine they'll get an exception which is correct.

Reply:
Done Thanks.